### ISSUE TITLE
[NBug] Unable to parse object ID "fatal: bad config line 40 in fil...

### ISSUE THREAD

**Issue by mx-bernhard** [gitextensions/gitextensions#6632]
## Current behaviour

gitextensions crashes.

## Expected behaviour

It probably should display an error in a message box that .gitconfig cannot be parsed properly - or specific parts of it cannot be parsed. 

## Steps to reproduce

edit .gitconfig. Add a line (or edit the existing line)

```
[core]
    editor = c:\foo.bat
```

Open gitextensions.

## Error Details
```
System.FormatException: Unable to parse object ID "fatal: bad config line 40 in file C:/Users/jahn/.gitconfig".
   at GitUIPluginInterfaces.ObjectId.Parse(String s)
   at GitCommands.GitModule.GetCurrentCheckout()
   at GitUI.RevisionGridControl.ForceRefreshRevisions()
   at GitUI.CommandsDialogs.FormBrowse.OnLoad(EventArgs e)
   at System.Windows.Forms.Form.OnCreateControl()
   at System.Windows.Forms.Control.CreateControl(Boolean fIgnoreVisible)
   at System.Windows.Forms.Control.CreateControl()
   at System.Windows.Forms.Control.WmShowWindow(Message& m)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.Form.WmShowWindow(Message& m)
   at ResourceManager.GitExtensionsFormBase.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
```

## Additional information
if my .gitconfig contains illegal backslashes followed by certain characters e. g. `\f` or `\g` in my editor = <value> line this crash happens when opening gitextensions. I prefer exiting gracefully with a meaningful message pointing me to the problem (the crash reporter does this to an extent, but not always).

## Environment
- Git Extensions 3.1.0.5877
- Build bb1687f6bd0450d0c1a2bae10a37c65493fc1294
- Git 2.21.0.windows.1
- Microsoft Windows NT 10.0.17763.0
- .NET Framework 4.7.3394.0
- DPI 96dpi (no scaling)

**Comment by johndog** [comment]
I stumbled over a similar one, involving Microsoft's GVFS:

System.FormatException: Unable to parse object ID "edef3e7c27a6b866e36f3e7494eea54e79abb5b1


A new version of GVFS is available. Run `gvfs upgrade` to start the upgrade.".
   at GitUIPluginInterfaces.ObjectId.Parse(String s)
   at GitCommands.GitModule.GetCurrentCheckout()
   at GitUI.CommandsDialogs.FormCheckoutBranch.OkClick()
   at GitUI.CommandsDialogs.FormCheckoutBranch.OkClick(Object sender, EventArgs e)
   at System.Windows.Forms.Control.OnClick(EventArgs e)
   at System.Windows.Forms.Button.OnClick(EventArgs e)
   at System.Windows.Forms.Button.OnMouseUp(MouseEventArgs mevent)
   at System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.ButtonBase.WndProc(Message& m)
   at System.Windows.Forms.Button.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)

**Comment by RussKie** [comment]
@johndog the stack trace looks the same, however the issue is different.
Please open a new issue, thanks

**Comment by gerhardol** [comment]
Duplicate of #6512 
Included in 3.2
